pwiz_tools/Skyline/SettingsUI/ViewLibraryDlg.cs

```diff
@@ -875,6 +937,35 @@ public void UpdateUI(bool selectionChanged = true)
             charge2Button.Checked = charge2Button.Enabled && Settings.Default.ShowCharge2;
         }
 
+        //Sets up the redundant dropdown menu and selects the spectrum to display on the graph
+        private SpectrumInfoLibrary setupComboBox(List<ComboOption> options, int index)
+        {
+            if (comboRedundantSpectra.SelectedItem != null && options.Contains(comboRedundantSpectra.SelectedItem))
+            {
+                return ((ComboOption)comboRedundantSpectra.SelectedItem).spectrumInfoLibrary;
+            }
+            else
+            {
+                comboRedundantSpectra.Items.Clear();
+                options.Sort();
+                foreach (var opt in options)
+                {
+                    comboRedundantSpectra.Items.Add(opt);
+                    if (opt.spectrumInfoLibrary.IsBest)
+                    {
+                        comboRedundantSpectra.SelectedIndexChanged -= redundantSpectrum_changed;

```

Comment:
Worth adding a comment here about what you are doing, i.e. avoid updating the UI.